core/templates/pages/exploration-player-page/layout-directives/audio-bar.component.spec.ts

```diff
@@ -0,0 +1,568 @@
+      'play button is clicked', () => {
+      let audioTranslation = {
+        en: Voiceover.createFromBackendDict({
+          filename: 'audio-en.mp3',
+          file_size_bytes: 0.5,
+          needs_update: false,
+          duration_secs: 0.5
+        }),
+        es: Voiceover.createFromBackendDict({
+          filename: 'audio-es.mp3',
+          file_size_bytes: 0.5,
+          needs_update: false,
+          duration_secs: 0.5
+        })
+      };
+      spyOn(audioTranslationManagerService, 'getCurrentAudioTranslations')
+        .and.returnValue(audioTranslation);
+      // Setting selected language to be 'en'.
+      spyOn(audioTranslationLanguageService, 'getCurrentAudioLanguageCode')
+        .and.returnValue('en');
+      // Setting auto generated langugae to be false.
+      spyOn(
+        audioTranslationLanguageService, 'isAutogeneratedLanguageCodeSelected')
+        .and.returnValue(false);
+      // Setting audio is playing to be true.
+      spyOn(audioPlayerService, 'isPlaying')
+        .and.returnValue(false);
+      // Settings audio tracks loaded to be false.
+      spyOn(audioPlayerService, 'isTrackLoaded').and.returnValue(false);
+      let loadAndPlaySpy = spyOn(component, 'loadAndPlayAudioTranslation')
+        .and.callThrough();
+      spyOn(playerPositionService, 'getCurrentStateName')
+        .and.returnValue('Start');
+      spyOn(audioPreloaderService, 'restartAudioPreloader')
+        .and.returnValue(null);
+
+      component.onPlayButtonClicked();
+      expect(loadAndPlaySpy).toHaveBeenCalled();
+    });
+
+    it('should pause uploaded audio translation when ' +
+      'pause button is clicked', () => {
+      let audioTranslation = {
+        en: Voiceover.createFromBackendDict({
+          filename: 'audio-en.mp3',
+          file_size_bytes: 0.5,
+          needs_update: false,
+          duration_secs: 0.5
+        }),
+        es: Voiceover.createFromBackendDict({
+          filename: 'audio-es.mp3',
+          file_size_bytes: 0.5,
+          needs_update: false,
+          duration_secs: 0.5
+        })
+      };
+      spyOn(audioTranslationManagerService, 'getCurrentAudioTranslations')
+        .and.returnValue(audioTranslation);
+      // Setting selected language to be 'en'.
+      spyOn(audioTranslationLanguageService, 'getCurrentAudioLanguageCode')
+        .and.returnValue('en');
+      // Setting auto generated langugae to be false.
+      spyOn(
+        audioTranslationLanguageService, 'isAutogeneratedLanguageCodeSelected')
+        .and.returnValue(false);
+      // Setting audio is playing to be true.
+      spyOn(audioPlayerService, 'isPlaying')
+        .and.returnValue(true);
+      let pauseSpy = spyOn(audioPlayerService, 'pause')
+        .and.returnValue(null);
+
+      component.onPlayButtonClicked();
+      expect(pauseSpy).toHaveBeenCalled();
+    });
+
+    it('should load audio track and play audio ' +
+      'play which are stored in cache', fakeAsync(() => {
```

Comment:
I think the description is wrong, should be `...play audio which are...`

Reply:
done